rust-src/concordium_base/src/id/types.rs

```diff
@@ -432,6 +432,25 @@ pub struct YearMonth {
     pub month: u8,
 }
 
+impl YearMonth {
+    /// Return the time at the beginning of the month.
+    pub fn lower(self) -> Option<chrono::DateTime<chrono::Utc>> {
+        let date = chrono::NaiveDate::from_ymd_opt(self.year.into(), self.month.into(), 1)?;
+        let time = chrono::NaiveTime::from_hms_opt(0, 0, 0)?;
+        let dt = date.and_time(time);
+        Some(chrono::DateTime::from_utc(dt, chrono::Utc))
+    }
+
+    /// Return the time at the beginning of the next month.
+    pub fn upper(self) -> Option<chrono::DateTime<chrono::Utc>> {
```

Comment:
In which context are these methods to be used?
I don't find their names very intuitive, but perhaps the context would help.

Reply:
They are used when passing policies to smart contracts. They are needed in the integration testing library.

They are used to supply the upper and lower bounds on the validity of the credential.

Reply:
Ok, thanks for adding the comments!